src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java

```diff
@@ -16,394 +16,559 @@
-	        "INSERT INTO " + TPCCConstants.TABLENAME_ORDERLINE + 
-	        " (OL_O_ID, OL_D_ID, OL_W_ID, OL_NUMBER, OL_I_ID, OL_SUPPLY_W_ID, OL_QUANTITY, OL_AMOUNT, OL_DIST_INFO) " +
-            " VALUES (?,?,?,?,?,?,?,?,?)");
-
-
-	// NewOrder Txn
-	private PreparedStatement stmtGetCust = null;
-	private PreparedStatement stmtGetWhse = null;
-	private PreparedStatement stmtGetDist = null;
-	private PreparedStatement stmtInsertNewOrder = null;
-	private PreparedStatement stmtUpdateDist = null;
-	private PreparedStatement stmtInsertOOrder = null;
-	private PreparedStatement stmtGetItem = null;
-	private PreparedStatement stmtGetStock = null;
-	private PreparedStatement stmtUpdateStock = null;
-	private PreparedStatement stmtInsertOrderLine = null;
-
-
-    public ResultSet run(Connection conn, Random gen,
-			int terminalWarehouseID, int numWarehouses,
-			int terminalDistrictLowerID, int terminalDistrictUpperID,
-			TPCCWorker w) throws SQLException {
-
-
-
-		//initializing all prepared statements
-		stmtGetCust=this.getPreparedStatement(conn, stmtGetCustSQL);
-		stmtGetWhse=this.getPreparedStatement(conn, stmtGetWhseSQL);
-		stmtGetDist=this.getPreparedStatement(conn, stmtGetDistSQL);
-		stmtInsertNewOrder=this.getPreparedStatement(conn, stmtInsertNewOrderSQL);
-		stmtUpdateDist =this.getPreparedStatement(conn, stmtUpdateDistSQL);
-		stmtInsertOOrder =this.getPreparedStatement(conn, stmtInsertOOrderSQL);
-		stmtGetItem =this.getPreparedStatement(conn, stmtGetItemSQL);
-		stmtGetStock =this.getPreparedStatement(conn, stmtGetStockSQL);
-		stmtUpdateStock =this.getPreparedStatement(conn, stmtUpdateStockSQL);
-		stmtInsertOrderLine =this.getPreparedStatement(conn, stmtInsertOrderLineSQL);
-
-
-		int districtID = TPCCUtil.randomNumber(terminalDistrictLowerID,terminalDistrictUpperID, gen);
-		int customerID = TPCCUtil.getCustomerID(gen);
-
-		int numItems = (int) TPCCUtil.randomNumber(5, 15, gen);
-		int[] itemIDs = new int[numItems];
-		int[] supplierWarehouseIDs = new int[numItems];
-		int[] orderQuantities = new int[numItems];
-		int allLocal = 1;
-		for (int i = 0; i < numItems; i++) {
-			itemIDs[i] = TPCCUtil.getItemID(gen);
-			if (TPCCUtil.randomNumber(1, 100, gen) > 1) {
-				supplierWarehouseIDs[i] = terminalWarehouseID;
-			} else {
-				do {
-					supplierWarehouseIDs[i] = TPCCUtil.randomNumber(1,
-							numWarehouses, gen);
-				} while (supplierWarehouseIDs[i] == terminalWarehouseID
-						&& numWarehouses > 1);
-				allLocal = 0;
-			}
-			orderQuantities[i] = TPCCUtil.randomNumber(1, 10, gen);
-		}
-
-		// we need to cause 1% of the new orders to be rolled back.
-		if (TPCCUtil.randomNumber(1, 100, gen) == 1)
-			itemIDs[numItems - 1] = TPCCConfig.INVALID_ITEM_ID;
-
-
-		newOrderTransaction(terminalWarehouseID, districtID,
-						customerID, numItems, allLocal, itemIDs,
-						supplierWarehouseIDs, orderQuantities, conn, w);
-		return null;
+  public final SQLStmt stmtUpdateStockSQL = new SQLStmt(
```

Comment:
something looks odd about indentation above (e.g. some lines not being changed, for others the highlighted spaces size&location differ)? If we are changing that can you double-check it is consistent. But probably easier to keep the existing style (or otherwise other files will now differ from the ones changed here).

Reply:
this should be solved.